### DISCUSSION TITLE
where does 'Auto groups by languages' get the languages from?

### DISCUSSION THREAD

**Discussion by michaelbeijer** [xiaoyifang/goldendict-ng#568] [Q&A] [answered]
Perhaps a stupid question, but where does **Auto groups by languages** get the languages from? From language codes added to the file name? Or inside the file? Like in a .dsl header, for example.  Or both?

Also, what rules must these language codes conform to? I noticed people have different ways of writing the language codes:

For example for Dutch-English, I have seen:

- Nld-Eng
- Nl-En
- nl-en

I'm trying to pick one and make all my dictionaries consistent. I personally prefer 'nl-en', and was considering standardising all my dictionaries using this system. That is: two-letter language codes, lowercase.

**Comment by michaelbeijer** [comment]
I just noticed that even if I name a dictionary like: "Medisch Woordenboek (**nl-nl**).dsl"

GoldenDict still uses **Nl-Nl** in its Groups interface. Is this the official format for writing these lang codes?

**Reply by michaelbeijer** [reply]
Thanks!

So GD never gets the language codes from the file names, but instead from inside the files themselves, right?

**Reply by xiaoyifang** [reply, marked as answer]
first try to get language from dictionary , if not found ,then try to get from filename.

**Reply by michaelbeijer** [reply]
Thanks, understood!

Is there any way for the user to customise whether GD represents languages like this: "En-En", or "en-en"? For example, in the Groups tab.

**Reply by xiaoyifang** [reply]
you mean group rename?

**Reply by michaelbeijer** [reply]
Oh yeah, of course, I hadn't thought of that. Of course, that would work too. The only problem is I would have to do it manually each time I re-create the groups, whereas it would be nice to be able to do it programmatically (once and for all).

**Reply by xiaoyifang** [reply]
group rename is already provided.

**Reply by michaelbeijer** [reply]
Yes, I know, but what I mean is I was looking for a way to change how the languages are shown, from the start. So, instead of GoldenDict's default way of displaying it being: "Nl - Nl", I would prefer to see "nl-nl". Instead of "En - En", I want: "en-en". No big deal though.

**Reply by xiaoyifang** [reply]
maybe wait for the metadata implementation.
